go/types/opcache.go

```diff
@@ -80,144 +147,185 @@ func (store *ldbOpCacheStore) opCache() opCache {
 	return &ldbOpCache{vrw: store.vrw, colId: colId, ldb: store.ldb}
 }
 
-// Set can be called from any goroutine
-func (opc *ldbOpCache) MapSet(mapKey Value, mapVal Value) {
-	mapKeyArray := [initialBufferSize]byte{}
-	mapValArray := [initialBufferSize]byte{}
-
-	switch mapKey.Type().Kind() {
-	default:
-		ldbKey := ldbKeyFromValueHash(mapKey, opc.colId)
-
-		// Since we've used the ref of keyValue as our ldbKey. We need to store mapKey and mapVal in the ldb value. We use the following format for that:
-		//
-		//     uint32 (4 bytes)             bytes                 bytes
-		// +-----------------------+---------------------+----------------------+
-		// | key serialization len |    serialized key   |   serialized value   |
-		// +-----------------------+---------------------+----------------------+
-
-		encodedMapKey := encToSlice(mapKey, mapKeyArray[:], opc.vrw)
-		encodedMapVal := encToSlice(mapVal, mapValArray[:], opc.vrw)
-		ldbValueArray := [initialBufferSize * 2]byte{}
-		binary.LittleEndian.PutUint32(ldbValueArray[:], uint32(len(encodedMapKey)))
-		ldbValue := ldbValueArray[0:4]
-		ldbValue = append(ldbValue, encodedMapKey...)
-		ldbValue = append(ldbValue, encodedMapVal...)
-
-		// TODO: Will manually batching calls to ldb.Put() help?
-		err := opc.ldb.Put(ldbKey, ldbValue, nil)
-		d.Chk.NoError(err)
-
-	case BoolKind, NumberKind, StringKind:
-		ldbKey := ldbKeyFromValue(mapKey, opc.colId, opc.vrw)
-		encodedMapVal := encToSlice(mapVal, mapValArray[:], opc.vrw)
-		err := opc.ldb.Put(ldbKey, encodedMapVal, nil)
-		d.Chk.NoError(err)
+// insertLdbOp encodes allKeys into the ldb key. Bool, Number, and String values
+// are encoded directly into the ldb key bytes. All other types are encoded as
+// there Hash() digest. Their actual value is then stored in ldb value.
+func (opc *ldbOpCache) insertLdbOp(allKeys ValueSlice, opKind NomsKind, val Value) {
+	d.Chk.True(len(allKeys) < 0x00FF, "Number of keys in GraphMapSet exceeds max of 256")
```

Comment:
Right. I merged that change into this branch a while back and must have missed this one. This line also had another bug. len(allKeys) can be <= 256. This changed to:
```    d.PanicIfTrue(len(allKeys) > 0x00FF```